main.py

```diff
@@ -0,0 +1,490 @@
+                        func()
+
+        super(Manager, self).process_events(event)
+
+
+class Window:
+    fonts = [
+        pygame.font.SysFont("monospace", 20),
+        pygame.font.SysFont("monospace", 24),
+        pygame.font.SysFont("arial", 18),
+    ]
+
+    cursors = [pygame.image.load("./assets/cursors/pointer.png"), pygame.image.load("./assets/cursors/crosshair.png")]
+    logo = pygame.transform.scale(pygame.image.load("./assets/DTIcon.png"), (100, 100))
+    clip = lambda x, a, b: a if x < a else b if x > b else x
+
+    @staticmethod
+    def renderText(surface, text, location):
+        whitetext = Window.fonts[2].render(text, 1, (255, 255, 255))
+        Window.fonts[0].set_bold(True)
+        blacktext = Window.fonts[2].render(text, 1, (0, 0, 0))
+        Window.fonts[0].set_bold(False)
+
+        textrect = whitetext.get_rect()
+        for i in range(-3, 4):
+            for j in range(-3, 4):
+                textrect.center = [a + b for a, b in zip(location, (i, j))]
+                surface.blit(blacktext, textrect)
+        textrect.center = location
+        surface.blit(whitetext, textrect)
+
+    def __init__(self, thermal_image=None, filename=None):
+        self.exthandler = WindowHandler()
+        if thermal_image is not None:
+            mat = thermal_image.thermal_np.astype(np.float32)
+        
+            if mat.shape != (512, 640):
+                y0, x0 = mat.shape
+                mat = zoom(mat, [512 / y0, 640 / x0])
+
+            self.mat = mat
+            self.mat_orig = mat.copy()
+            self.mat_emm = mat.copy()
+            self.raw = thermal_image.raw_sensor_np
+            self.meta = thermal_image.meta
+            self.overlays = pygame.Surface((640, 512), pygame.SRCALPHA)
+        else:
+            with open(filename, "rb") as f:
+                data = pickle.load(f)
+            self.mat = data.mat
+            self.mat_orig = data.mat_orig
+            self.mat_emm = data.mat_emm
+            self.raw = data.raw
+            self.meta = data.meta
+            self.overlays = pygame.image.fromstring(data.overlays, (640, 512), "RGBA")
+
+            for entry in data.tableEntries:
+                self.exthandler.addToTable(entry)
+            self.exthandler.loadGraph(data.plots)
+
+        self.colorMap = "jet"
+        self.lineNum = 0
+        self.boxNum = 0
+        self.spotNum = 0
+        self.areaMode = "poly"
+        self.selectionComplete = False
+        self.work("colorMap", self.colorMap)
+
+        self.mode = "main"
+        # Dictionary of pages. Each page is a manager.
+        self.managers = {}
+        self.managers["main"] = Manager(
+            buttons=[
+                ((15, 15), (215, 45), "Spot marking", lambda: self.changeMode("spot")),
+                ((15, 75), (215, 45), "Line measurement", lambda: self.changeMode("line")),
+                ((15, 135), (215, 45), "Area marking", lambda: self.changeMode("area")),
```

Comment:
In the Area marking mode, if I accidentally don't drag and draw, but just mark a point, the following error comes. Can you catch this? (Probably zero area rect)

```
$ python main.py
pygame 2.0.1 (SDL 2.0.14, Python 3.7.4)
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "main.py", line 482, in <module>
    window.process(event)
  File "main.py", line 352, in process
    self.managers[self.mode].process_events(event)
  File "main.py", line 47, in process_events
    func()
  File "main.py", line 149, in <lambda>
    ((15, 470), (215, 45), "Continue", lambda: self.work("area")),
  File "main.py", line 275, in work
    self.exthandler.addToTable([f"a{self.boxNum}", np.min(chunk), np.max(chunk), np.mean(chunk)])
  File "<__array_function__ internals>", line 6, in amin
  File "\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py", line 2880, in amin
    keepdims=keepdims, initial=initial, where=where)
  File "\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py", line 86, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
ValueError: zero-size array to reduction operation minimum which has no identity
```